src/jupyter_contrib_nbextensions/nbextensions/execution_dependencies/execution_dependencies.js

```diff
@@ -0,0 +1,96 @@
+                            dep_graph[id] = deps;
+                        });
+                    });
+
+                    console.log('Collect in-degrees...');
+                    var in_degree = {};                                                                             // ...collect in-degrees of nodes
+                    for(var key in dep_graph) {
+                        for (var i=0,  tot=dep_graph[key].length; i < tot; i++) {
+                            var dep = dep_graph[key][i];
+                            in_degree[key]   = in_degree[key] || 0;
+                            in_degree[dep] = in_degree[dep] === undefined ? 1 : ++in_degree[dep];
+                        }
+                    }
+
+                    console.log('Fill processing queue...');
+                    var processing_queue = [];                                                                      // ...add all nodes with in-degree 0 to queue
+                    for(var key in dep_graph) {
+                        if(in_degree[key] == 0) {
+                            processing_queue.push(key);
+                        }
+                    }
+
+                    console.log('Start topological sort...');
+                    var processed_nodes = 0;                                                                        // ...number of processed nodes (to detect circular dependencies)
+                    var processing_order = [];
+
+                    while(processing_queue.length > 0 && processed_nodes < Object.keys(dep_graph).length) {         // ...stay processing deps while the queue contains nodes and the processed nodes are below total node quantity
+                        console.log('Processing queue: ', processing_queue);
+                        console.log('Processing order: ', processing_order);
+                        var id = processing_queue.shift();                                                          // .....pop front of queue and front-push it to the processing order
+                        processing_order.unshift(id);
+                        console.log('Process node:', id);
+
+                        for (var i=0,  tot=dep_graph[id].length; i < tot; i++) {                                   // ......iterate over dependent nodes of current id and decrease their in-degree by 1
+                            var dep = dep_graph[id][i];
+                            in_degree[dep]--;
+                            if(in_degree[dep] == 0) {                                                               // ......queue dependency if in-degree is 0
+                                processing_queue.unshift(dep);
+                            }
+                        }
+                        processed_nodes++;
+                    }
+
+                    if(processed_nodes >= Object.keys(dep_graph).length) {
+                        console.error('There is a circular dependency in your execute dependencies!');
```

Comment:
you could do an `alert` here, or alternatively use the notebook dialog for something prettier. See https://github.com/ipython-contrib/jupyter_contrib_nbextensions/blob/a4544eca1182db2816523959002b5a31c263e267/src/jupyter_contrib_nbextensions/nbextensions/init_cell/main.js#L106-L115 for an example usage.

In fact, it probably makes sense for this extension to refuse to execute dependencies in untrusted notebooks in the same way that init_cell does...

Reply:
Sounds great, I will add it.